### PR TITLE
Implement Navidrome sharing

### LINKED ISSUE #80: Share from context menu
## What do you want to be added?
Navidrome has recently implemented sharing! It would be convenient to interact with this feature through Feishin, rather than navigating to my Navidrome instance.

## Additional context
https://github.com/navidrome/navidrome/pull/2106

### PR DESCRIPTION
**Pull request by iiPythonx**
PR that adds an option for sharing an album or songs via Navidrome (nothing else yet), resolves #80.  
There are more then likely *a lot* of issues with how I did this, if you have any feedback on it please lmk (considering I've still barely touched React in my life).

Two nitpicks:
- Right now, this PR selects `resourceType` based on the first item you've selected. This *should* be fine since I'm assuming you can't magically select two items of different types within Feishin (ie. selecting an album <ins>and</ins> a song at the same time).
- Additionally, there seems to be a lack of margin on TextInput fields (not that bad but it's a bit squished):

image

Anyways, as stated in the beginning, any feedback would be very much appreciated.

### CODE REVIEW

File: src/renderer/api/controller.ts [hunk 1]
```diff
@@ -223,6 +228,7 @@ const endpoints: ApiController = {
         scrobble: ssController.scrobble,
         search: ssController.search3,
         setRating: undefined,
+        shareItem: undefined,
```
Comment (kgarner7) [inline src/renderer/api/controller.ts:231]: Strictly speaking (even though it wouldn't be used for now) Subsonic API also supports sharing https://subsonic.org/pages/api.jsp#createShare

File: src/renderer/api/navidrome/navidrome-types.ts [hunk 2]
```diff
@@ -343,6 +343,17 @@ const removeFromPlaylistParameters = z.object({
     id: z.array(z.string()),
 });
 
+const shareItem = z.object({
+    id: z.string(),
+});
+
+const shareItemParameters = z.object({
+    description: z.string(),
+    downloadable: z.boolean(),
```
Comment (kgarner7) [inline src/renderer/api/navidrome/navidrome-types.ts:352]: Navidrome (and subsonic) also supports `expires` value.

File: src/renderer/api/types.ts [hunk 3]
```diff
@@ -766,6 +766,18 @@ export type RatingQuery = {
 
 export type SetRatingArgs = { query: RatingQuery; serverId?: string } & BaseEndpointArgs;
 
+// Sharing
+export type ShareItemResponse = { id: string } | undefined;
+
+export type ShareItemBody = {
+    description: string;
+    downloadable: boolean;
+    resourceIds: string;
```
Comment (kgarner7) [inline src/renderer/api/types.ts:776]: Similar, should probably also have expiry here as well (optional, either date or unix timestamp)

**Review by kgarner7 — CHANGES_REQUESTED** [review]
Sharing is not available in every Navidrome. Make sure to add it as a feature in `feature-types.ts` (ServerFeature), modify `navidrome-controllers.ts#getServerInfo` to check for the correct version, and check if that feature exists.

File: src/renderer/api/navidrome/navidrome-types.ts (hunk 2 above)
Comment (iiPythonx) [reply src/renderer/api/navidrome/navidrome-types.ts:352]: I'll get to work on implementing this right now along with the server version check.

**Review by kgarner7 — CHANGES_REQUESTED** [review]
I added a few changes. It might be nice to have the ability to view existing shares, but this is a lower priority (I _may_ take a look at implementing this).

I'm not particularly a fan of having the share only copied to clipboard, as that might not work especially in browser environments where clipboard access is limited. It would be good to have a way to open in browser (maybe when clicking on the notification?)

**Comment by iiPythonx** [comment]
> It would be good to have a way to open in browser (maybe when clicking on the notification?)

Went ahead and made it so clicking on the success notification does indeed open the share in browser.  

> It might be nice to have the ability to view existing shares, but this is a lower priority (I may take a look at implementing this).

I'll leave that up to you then (if you decide to work on it).

**Review by kgarner7 — APPROVED** [review]
...ngl, adding share listing is more of a pain than I'd like. It can come as a feature later.

**Comment by iiPythonx** [comment]
Replacement is at https://github.com/jeffvli/feishin/pull/575 (sorry!).

**Pull request closed without merging**